### PR TITLE
Fix clippy warning related to IString PartialOrd implementation

### PR DESCRIPTION
**Pull request by ColonelThirtyTwo**

### CODE REVIEW

File: src/string.rs [hunk 1]
```diff
@@ -125,8 +125,12 @@ impl Ord for IString {
         Ord::cmp(AsRef::<str>::as_ref(self), AsRef::<str>::as_ref(other))
     }
 }
```
Comment (kirillsemyonkin) [inline src/string.rs:127]: why doesnt cargo fmt test check for missing newlines inbetween items

**Comment by kirillsemyonkin** [comment]
Please provide description of why is this needed. I'm being more curious than strict, as to why it passed before and why it needs to be written out like this now.

File: src/string.rs [hunk 2]
```diff
@@ -125,8 +125,12 @@ impl Ord for IString {
...
+impl PartialOrd for IString {
```
Comment (kirillsemyonkin) [inline src/string.rs:128]: a description for this change should probably also be reflected in a code comment

**Comment by ColonelThirtyTwo** [comment]
> Please provide description of why is this needed. I'm being more curious than strict, as to why it passed before and why it needs to be written out like this now.

As can be seen in the CI failures for my other PRs, the version of clippy included in Rust 1.73.0 issues an error due to `PartialOrd<IString> for IString` not using the `Ord` implementation:

```
> cargo clippy
    Checking implicit-clone v0.4.1 (/home/col/Projects/Code/Libraries/implicit-clone)
error: incorrect implementation of `partial_cmp` on an `Ord` type
   --> src/string.rs:103:9
    |
103 | /          impl $trait<$type2> for $type1 {
104 | |              fn $fn(&self, other: &$type2) -> $ret {
    | | ___________________________________________________-
105 | ||                 $trait::$fn(AsRef::<str>::as_ref(self), AsRef::<str>::as_ref(other))
106 | ||             }
    | ||_____________- help: change this to: `{ Some(self.cmp(other)) }`
107 | |          }
    | |__________^
...
129 |    impl_cmp_as_str!(PartialOrd::<IString, IString>);
    |    ------------------------------------------------ in this macro invocation
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#incorrect_partial_ord_impl_on_ord_type
    = note: `#[deny(clippy::incorrect_partial_ord_impl_on_ord_type)]` on by default
    = note: this error originates in the macro `impl_cmp_as_str` (in Nightly builds, run with -Z macro-backtrace for more info)

error: incorrect implementation of `partial_cmp` on an `Ord` type
   --> src/string.rs:103:9
    |
103 | /          impl $trait<$type2> for $type1 {
104 | |              fn $fn(&self, other: &$type2) -> $ret {
    | | ___________________________________________________-
105 | ||                 $trait::$fn(AsRef::<str>::as_ref(self), AsRef::<str>::as_ref(other))
106 | ||             }
    | ||_____________- help: change this to: `{ Some(self.cmp(other)) }`
107 | |          }
    | |__________^
...
129 |    impl_cmp_as_str!(PartialOrd::<IString, IString>);
    |    ------------------------------------------------ in this macro invocation
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#incorrect_partial_ord_impl_on_ord_type
    = note: this error originates in the macro `impl_cmp_as_str` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `implicit-clone` (lib) due to 2 previous errors
```

File: src/string.rs (hunk 2 above)
Comment (ColonelThirtyTwo) [reply src/string.rs:131]: Done

File: src/string.rs (hunk 1 above)
Comment (ColonelThirtyTwo) [reply src/string.rs:127]: I've never seen rustfmt care about newlines between items. Regardless, I've added one.

Comment (kirillsemyonkin) [reply src/string.rs:127]: yes, it doesnt care, and thats what im complaining about lmao
you good tho

**Pull request merged**